### PR TITLE
Stop processing the pull-request if Github fails

### PR DESCRIPTION
**Pull request by apelisse**
Today we keep going because we don't properly handle errors, and we may
end-up posting a comment everytime we can't read the comments.
Let's check if we can fetch the comment and fail if it's broken.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/1597)

### CODE REVIEW

**Comment by apelisse** [comment]
@lavalamp this is an attempt to fix the bug on https://github.com/kubernetes/kubernetes/pull/27193.

I'm not sure if this is what went wrong, but that can't hurt anyway.

File: mungegithub/mungers/close-stale-pr.go [hunk 1]
```diff
@@ -295,7 +303,7 @@ func (CloseStalePR) Munge(obj *github.MungeObject) {
 		checkAndWarn(obj, inactiveFor, closeIn)
 	} else {
 		// Pull-request is active. Remove previous potential warning
-		comment := findLatestWarningComment(obj)
+		comment, _ := findLatestWarningComment(obj)
```
Comment (lavalamp) [inline mungegithub/mungers/close-stale-pr.go]: handle the error.

File: mungegithub/mungers/close-stale-pr.go [hunk 2]
```diff
@@ -180,12 +180,12 @@ func findLastModificationTime(obj *github.MungeObject) (*time.Time, error) {
 	return lastModif, nil
 }
 
-func findLatestWarningComment(obj *github.MungeObject) *githubapi.IssueComment {
+func findLatestWarningComment(obj *github.MungeObject) (*githubapi.IssueComment, error) {
```
Comment (lavalamp) [inline mungegithub/mungers/close-stale-pr.go]: nit: document, include that `nil, nil` is a possible return value.

**Pull request merged**